pyexchange/korbit.py

```diff
@@ -0,0 +1,374 @@
+    """Korbit API interface.
+
+    Developed according to the following manual:
+    <https://apidocs.korbit.co.kr/>.
+    
+    Authentication uses OAuth 2.0. Access tokens expire within one hour, 
+    and requires refresh token to be called periodically
+    """
+
+    logger = logging.getLogger()
+
+    def __init__(self, api_server: str, api_key: str, secret_key: str, timeout: float):
+        assert(isinstance(api_key, str))
+        assert(isinstance(secret_key, str))
+
+        self.api_server = api_server
+        self.api_key = api_key
+        self.secret_key = secret_key
+        self.timeout = timeout
+        self.token = {}
+        self.time_to_expiry = 0
+
+    def get_balances(self):
+        return self._http_authenticated_request("GET", "/v1/user/balances", {})
+
+    def get_markets(self):
+        return self._http_unauthenticated_request("GET", "/v1/ticker/detailed/all", {})
+
+    def get_pair(self, pair: str):
+        assert(isinstance(pair, str))
+        return self.get_markets()[f"{pair}"]
+
+    # List keepers open orders
+    def get_orders(self, pair: str) -> List[Order]:
+        assert(isinstance(pair, str))
+
+        orders = self._http_authenticated_request("GET", f"/v1/user/orders/open?currency_pair={pair}", {})
+        return list(map(lambda item: Order.from_list(item, pair), orders))
+
+    def place_order(self, pair: str, is_sell: bool, price: int, amount: Wad) -> str:
+        assert(isinstance(pair, str))
+        assert(isinstance(is_sell, bool))
+        assert(isinstance(price, int))
+        assert(isinstance(amount, Wad))
+
+        side = "buy" if is_sell == False else "sell"
+        nonce = int(time.time()) # Nonce must be monotonically increasing
+        
+        data = {
+            "currency_pair": pair,
+            "type": "limit",
+            "price": int(price),
+            "coin_amount": str(amount),
+            "nonce": nonce
+        }
+
+        self.logger.info(f"Placing order ({side}, amount {data['coin_amount']} of {pair},"
+                         f" price {data['price']})...")
+        result = self._http_authenticated_request("POST", f"/v1/user/orders/{side}", data)
+        order_id = result['orderId']
+
+        self.logger.info(f"Placed order (#{result}) as #{order_id}")
+
+        return order_id
+
+    def cancel_order(self, order_id: int, pair: str) -> bool:
+        assert(isinstance(order_id, int))
+        assert(isinstance(pair, str))
+
+        self.logger.info(f"Cancelling order #{order_id}...")
+        
+        nonce = int(time.time()) # Nonce must be monotonically increasing
+
+        data = {
+            "currency_pair": pair,
+            "nonce": nonce,
+            "id": order_id
+        }
+
+        result = self._http_authenticated_request("POST", f"/v1/user/orders/cancel", data)
+        if result[0]["status"] == "success":
+            return True
+        else:
+            return False
+
+    # TODO: finish implementation
```

Comment:
Is this a stale comment, or is something actually incomplete?

Reply:
Stale comment, I will remove it